### PR TITLE
xenPackages: deprecate Xen 4.5, security support ended

### LINKED ISSUE #36436: xen 4.5 end-of-support, should be deprecated
## Issue description

Security support for Xen 4.5 (our current default version) ended January 2018,
so we should change the default to 4.8 in 18.03 (and remove 4.5 in the future).

### PR DESCRIPTION
**Pull request by oxij**
###### Motivation for this change

The title. Closes  #36436.

###### Things done

- [X] It builds.
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by oxij** [comment]
@GrahamcOfBorg build xenPackages.xen_4_5-light

@GrahamcOfBorg build xenPackages.xen_4_8-light

**Comment by oxij** [comment]
@grahamc these ofborg results look a bit strange to me. Why is "not supported" an error while "vulnerable" is simply refused?

**Comment by oxij** [comment]
Tested to build and give proper errors. LGTM.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -18353,9 +18353,9 @@ with pkgs;
 
   xenPackages = recurseIntoAttrs (callPackage ../applications/virtualization/xen/packages.nix {});
 
-  xen = xenPackages.xen_4_5-vanilla;
-  xen-slim = xenPackages.xen_4_5-slim;
-  xen-light = xenPackages.xen_4_5-light;
+  xen = xenPackages.xen_4_8-vanilla;
```
Comment (7c6f434c) [inline pkgs/top-level/all-packages.nix]: Any reason it is not a version-less reference to `xenPackages.xen-vanilla`?

**Comment by 7c6f434c** [comment]
I hope that these checks were done by different revisions of ofborg — it is in flux right now re: instantiation failure handling.

**Comment by oxij** [comment]
> Any reason it is not a version-less reference to `xenPackages.xen-vanilla`?

Fixed.

**Comment by xeji** [comment]
This change should go into 18.03 as well but cannot simply be cherry-picked (because we introduced xen 4.10 on master). I will port it in a separate pull request.

**Comment by oxij** [comment]
Btw,
@xeji thanks for reminding me about this
@7c6f434c thanks for reviewing this

Should I make a PR to move default `xen` from `vanilla` to `small` (that was the original plan, because this way we are going to be less out of date on qemu vulns) or are there any counterarguments to this? That would probably break some configs.
/cc @grahamc on this

**Comment by xeji** [comment]
Moving to `slim` (guess that's what you mean) sounds good - we'll get upstream qemu fixes earlier. One caveat is that the Xen project claims to add some extra fixes: 

>  The qemu-xen code is maintained upstream in the qemu.org git trees. In addition the Xen project also maintains its own stable branch of qemu, based on the upstream stable branches with a small number of additional fixes for Xen. 
 
Don't know about the number or importance of such additional fixes.

**Pull request merged**